packages/agent/src/certificate.ts

```diff
@@ -59,19 +74,38 @@ export function hashTreeToString(tree: HashTree): string {
     case NodeId.Empty:
       return '()';
     case NodeId.Fork: {
-      const left = hashTreeToString(tree[1]);
-      const right = hashTreeToString(tree[2]);
-      return `sub(\n left:\n${indent(left)}\n---\n right:\n${indent(right)}\n)`;
+      if (tree[1] instanceof Array && tree[2] instanceof ArrayBuffer) {
+        const left = hashTreeToString(tree[1]);
+        const right = hashTreeToString(tree[2]);
+        return `sub(\n left:\n${indent(left)}\n---\n right:\n${indent(right)}\n)`;
+      } else {
+        throw new Error('Invalid tree structure for fork');
+      }
     }
     case NodeId.Labeled: {
-      const label = labelToString(tree[1]);
-      const sub = hashTreeToString(tree[2]);
-      return `label(\n label:\n${indent(label)}\n sub:\n${indent(sub)}\n)`;
+      if (tree[1] instanceof ArrayBuffer && tree[2] instanceof ArrayBuffer) {
```

Comment:
some links to the spec as a comment would be helpful here imo

Reply:
This entire utility function doesn't seem to work and it has no tests. I only modified it in order to make it type safe with the other lookup changes